### PR TITLE
Follow up fixes for multiple muffler hatches.

### PR DESCRIPTION
**Pull request by AbdielKavash**
This is a follow up to https://github.com/GTNewHorizons/GT5-Unofficial/pull/3489 fixing several problems with that PR.

* ~~Logic for pollution output no longer creates unnecessary copies of arrays, instead it picks a random index into the existing list of muffler hatches.~~ See comment below.
* Said logic is also simplified in case the multi has 0 or 1 muffler hatches (as a large majority of multis in a typical base will) to avoid hatch selection completely.
* Removed unnecessary array copy when generating info data.
* Info data now mentions and correctly displays average pollution output also for all other machines that override `getInfoData`.
* EBF and MEBF now calculate pollution gas output based on the average pollution output of all mufflers (this *currently* does not matter, as both require exactly 1 muffler, but it is calculated correctly just in case).

### CODE REVIEW

**Comment by AbdielKavash** [comment]
Decided to just make this work correctly instead of the RNG workaround, don't merge yet please.

**Comment by AbdielKavash** [comment]
New code now spreads the pollution to be output evenly across all mufflers, instead of picking them at random. This means that a single muffler can now output less than 10,000 units of pollution at a time.

A machine will only output up to 10,000 units of pollution per muffler per tick. This is consistent with previous behavior; and as of right now entirely irrelevant, as no machine actually generates enough pollution that this becomes a meaningful bottleneck, unless one changes their configs.

In order to keep Advanced Muffler filter consumption at the same rate as it was previously, if an Advanced Muffler outputs less than 10,000 units of pollution, it has a (linearly proportional) chance to not damage its filter. Thus, for example, an XLGT with four Advanced Mufflers will attempt to output 2,500 units of pollution from each muffler, causing a 25% chance of damaging each of the filters.

This works correctly even when a combination of regular and advanced mufflers of various tiers is used, for the single person out there crazy enough to do this.

One very small breaking change introduced in this PR: if a multiblock has multiple muffler hatches, and one of them is obstructed, previously the machine would keep running if the only (first in the list) muffler that was used to output pollution was free. With this change, since all mufflers are used to output pollution evenly, if any muffler is blocked the machine will stall. Therefore the player needs to ensure that all mufflers are facing air at all times. This is now the intended functionality, the previous behavior is considered a bug.

**Comment by AbdielKavash** [comment]
Ready for review now.

File: src/main/java/gregtech/api/metatileentity/implementations/MTEMultiBlockBase.java [hunk 1]
```diff
@@ -632,23 +632,73 @@ protected void runMachine(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
...
+            int mufflerCount = 0;
+            int ventAmount = 0; // Allow venting of up to VENT_AMOUNT of pollution per muffler.
+            for (MTEHatchMuffler muffler : validMTEList(mMufflerHatches)) {
+                mufflerCount++;
+                if (ventAmount + VENT_AMOUNT <= mPollution) {
+                    ventAmount += VENT_AMOUNT;
+                }
+            }
+            // This might lose some small amount of pollution due to rounding, this is fine.
+            ventAmount /= mufflerCount;
```
Comment (RecursivePineapple) [inline src/main/java/gregtech/api/metatileentity/implementations/MTEMultiBlockBase.java:669]: btw, you can use `GTUtility.ceilDiv` for situations where rounding too low matters

**Review by RecursivePineapple — APPROVED** [review]

**Pull request merged**